dsl_parser/functions.py

```diff
@@ -592,6 +547,103 @@ def _validate_ref(self, ref, ref_name):
                                        self.attribute_path))
 
 
+def _get_attribute_from_node_instance(ni, node, attribute_path, path):
+    value = _get_property_value(ni['node_id'],
+                                ni.get('runtime_properties'),
+                                attribute_path,
+                                path,
+                                raise_if_not_found=False)
+    if value is None:
+        # attribute not found in instance runtime properties
+        if value is None:
+            # special case for { get_attributes_list: [
+            #                        ..., node_instance_id] }
+            # returns the node-instance-id
+            if len(attribute_path) == 1\
+                    and attribute_path[0] == 'node_instance_id':
+                value = ni['id']
```

Comment:
should this be here, or before checking the runtime props?

Reply:
It should be there- it appears we made it a pseudo-runtime property.

This means we go:
1. Is it in the runtime properties? If yes, use it.
2. Otherwse: Is it called node_instance_id? Then return the id of the node_instance.
3. Otherwise: If it's in node properties, return it
4. Otherwise, return nothing